control-plane-operator/controllers/hostedcontrolplane/v2/cno/component.go

```diff
@@ -100,3 +102,65 @@ func SetRestartAnnotationAndPatch(ctx context.Context, crclient client.Client, d
 
 	return nil
 }
+
+type operand struct {
+	DeploymentName  string
+	ContainerName   string
+	ReleaseImageKey string
+}
+
+var (
+	operandsDeploymentsList = []operand{
+		{
+			DeploymentName:  "ovnkube-control-plane",
+			ContainerName:   "ovnkube-control-plane",
+			ReleaseImageKey: "ovn-kubernetes",
+		},
+		{
+			DeploymentName:  "network-node-identity",
+			ContainerName:   "approver",
+			ReleaseImageKey: "ovn-kubernetes",
+		},
+		{
+			DeploymentName:  "multus-admission-controller",
+			ContainerName:   "multus-admission-controller",
+			ReleaseImageKey: "multus-admission-controller",
+		},
+		{
+			DeploymentName:  "cloud-network-config-controller",
+			ContainerName:   "controller",
+			ReleaseImageKey: "cloud-network-config-controller",
+		},
+	}
+)
+
+func checkOperandsRolloutStatus(cpContext component.WorkloadContext) (bool, error) {
```

Comment:
we've got code that does a similar thing in the hypershift install cli subcommand, maybe you can reuse?
https://github.com/openshift/hypershift/blob/main/cmd/install/install.go#L405